### ISSUE TITLE
blank screen on Contacts app since last update (4.0)

### ISSUE THREAD

**Issue by olivier6931** [nextcloud/contacts#2338]
Hello guys,

To begin, thanks to all for your work!

So... my issue :
Since the last upgrade of the app (to 4.0.0), I got a blank screen on Contacts page on my Nextcloud instance.
Contact sync with davX seems to work fine, but the frontend stay blank, as follow :

![image](https://user-images.githubusercontent.com/67951376/124732674-b5a82380-df13-11eb-8595-98e0bd329fbe.png)

Thanks for your help !

### Server configuration

**Operating system**: Linux Debian 10 + docker (CE) 20.10.7

**Web server:** Apache (based on image "nextcloud:latest") + reverse proxy Nginx

**Database:** MariaDB (based on image "mariadb:latest")

**PHP version:** PHP 7.4.2 (based on image "nextcloud:latest")

**Nextcloud version:** 21.0.3 (based on image "nextcloud:latest")

**Contacts version:**  4.0.0

**Updated from an older Nextcloud or fresh install:** updated (with docker compose + local volumes)

**Signing status:** No errors have been found.

**List of activated apps:**
  - accessibility: 1.7.0
  - bookmarks: 4.2.2
  - bruteforcesettings: 2.2.0
  - calendar: 2.3.0
  - cloud_federation_api: 1.4.0
  - contacts: 4.0.0
  - cookbook: 0.8.4
  - dashboard: 7.1.0
  - dav: 1.17.1
  - deck: 1.4.2
  - federatedfilesharing: 1.11.0
  - files: 1.16.0
  - files_external: 1.12.0
  - files_rightclick: 1.0.0
  - files_trashbin: 1.11.0
  - lookup_server_connector: 1.9.0
  - notes: 4.1.0
  - notifications: 2.9.0
  - oauth2: 1.9.0
  - passwords: 2021.7.21
  - provisioning_api: 1.11.0
  - settings: 1.3.0
  - theming: 1.12.0
  - twofactor_backupcodes: 1.10.0
  - updatenotification: 1.11.0
  - viewer: 1.5.0
  - weather_status: 1.1.0
  - workflowengine: 2.3.0

### Client configuration

**Browser:** Firefox 89 (desktop & mobile) or edge 91
**Operating system:** Windows 10 + Android 11

#### *Web server error log*
no special errors

#### *data/nextcloud.log*
no special errors

#### Browser log
# Firefox
Uncaught TypeError: OC.Share is undefined
    87293 constants.ts:128
    Webpack 6
constants.ts:128:16
    87293 constants.ts:128
    Webpack 6

# Edge
constants.ts:128 Uncaught TypeError: Cannot read property 'SHARE_TYPE_USER' of undefined
    at Module.87293 (constants.ts:128)
    at a (bootstrap:19)
    at Object.46278 (index.js:27)
    at a (bootstrap:19)
    at main.js:29
    at contacts-main.js?v=cd5df8c9-1:2
    at contacts-main.js?v=cd5df8c9-1:2
client.js:24 Uncaught (in promise) RuntimeError: abort(CompileError: WebAssembly.instantiate(): Wasm code generation disallowed by embedder). Build with -s ASSERTIONS=1 for more info.
    at nt (client.js:24)
    at client.js:24

**Comment by olivier6931** [comment]
Filling the issue, I noticed a thing that led me to check my other applications (and to be more precise, all sharings ones)
In the end, activating back the app "File sharing" solved my problem...

The good news is that the form to open an issue is really well done, and helped me.
The bad is I can't read log... shame on me! :)

Anyways, thanks again for your work!

**Comment by skjnldsv** [comment]
Reopening, we need to take those issues into account.
@ChristophWurst we use the sharing API for autocompletion for circles, any idea what we shall do if files_sharing is disabled?

**Comment by ChristophWurst** [comment]
One can disable sharing? I didn't know. I guess then those constants should be moved somewhere independent so they can be used despite the sharing enabled/disabled state.

**Comment by skjnldsv** [comment]
Even that, the autocomplete/search api would not be available to add members

**Comment by szaimen** [comment]
Another solution would be disallowing the sharing app to get disabled but would be a kind of taking this feature away for some users of course...

**Comment by RlndVt** [comment]
I'm having the same issue. For me sharing was enabled, I tried toggling without effect. Also tried enabling and disabling circles.

No entries in nextcloud.log. Anything else I could add?

```
$ ./occ app:list
Enabled:
  - accessibility: 1.7.0
  - activity: 2.14.3
  - admin_audit: 1.11.0
  - bruteforcesettings: 2.2.0
  - calendar: 2.3.0
  - checksum: 1.1.2
  - cloud_federation_api: 1.4.0
  - comments: 1.11.0
  - contacts: 4.0.0
  - contactsinteraction: 1.2.0
  - cookbook: 0.9.0
  - dashboard: 7.1.0
  - dav: 1.17.1
  - extract: 1.3.1
  - federatedfilesharing: 1.11.0
  - federation: 1.11.0
  - files: 1.16.0
  - files_pdfviewer: 2.1.0
  - files_retention: 1.10.1
  - files_rightclick: 1.0.0
  - files_sharing: 1.13.1
  - files_trashbin: 1.11.0
  - files_versions: 1.14.0
  - files_videoplayer: 1.10.0
  - firstrunwizard: 2.10.0
  - logreader: 2.6.0
  - lookup_server_connector: 1.9.0
  - nextcloud_announcements: 1.10.0
  - notes: 4.1.0
  - notifications: 2.9.0
  - oauth2: 1.9.0
  - password_policy: 1.11.0
  - phonetrack: 0.6.7
  - photos: 1.3.0
  - privacy: 1.5.0
  - provisioning_api: 1.11.0
  - quota_warning: 1.10.0
  - ransomware_protection: 1.10.1
  - recommendations: 1.0.0
  - registration: 1.3.0
  - serverinfo: 1.11.0
  - settings: 1.3.0
  - sharebymail: 1.11.0
  - socialsharing_email: 2.2.0
  - support: 1.4.0
  - survey_client: 1.9.0
  - systemtags: 1.11.0
  - tasks: 0.13.6
  - text: 3.2.0
  - theming: 1.12.0
  - twofactor_backupcodes: 1.10.0
  - updatenotification: 1.11.0
  - user_status: 1.1.1
  - viewer: 1.5.0
  - weather_status: 1.1.0
  - workflowengine: 2.3.0
```

**Comment by olivier6931** [comment]
> 
> 
> I'm having the same issue. For me sharing was enabled, I tried toggling without effect. Also tried enabling and disabling circles.
> 
> No entries in nextcloud.log. Anything else I could add?
> 
> ```
> $ ./occ app:list
> Enabled:
>   - accessibility: 1.7.0
>   - activity: 2.14.3
>   - admin_audit: 1.11.0
>   - bruteforcesettings: 2.2.0
>   - calendar: 2.3.0
>   - checksum: 1.1.2
>   - cloud_federation_api: 1.4.0
>   - comments: 1.11.0
>   - contacts: 4.0.0
>   - contactsinteraction: 1.2.0
>   - cookbook: 0.9.0
>   - dashboard: 7.1.0
>   - dav: 1.17.1
>   - extract: 1.3.1
>   - federatedfilesharing: 1.11.0
>   - federation: 1.11.0
>   - files: 1.16.0
>   - files_pdfviewer: 2.1.0
>   - files_retention: 1.10.1
>   - files_rightclick: 1.0.0
>   - files_sharing: 1.13.1
>   - files_trashbin: 1.11.0
>   - files_versions: 1.14.0
>   - files_videoplayer: 1.10.0
>   - firstrunwizard: 2.10.0
>   - logreader: 2.6.0
>   - lookup_server_connector: 1.9.0
>   - nextcloud_announcements: 1.10.0
>   - notes: 4.1.0
>   - notifications: 2.9.0
>   - oauth2: 1.9.0
>   - password_policy: 1.11.0
>   - phonetrack: 0.6.7
>   - photos: 1.3.0
>   - privacy: 1.5.0
>   - provisioning_api: 1.11.0
>   - quota_warning: 1.10.0
>   - ransomware_protection: 1.10.1
>   - recommendations: 1.0.0
>   - registration: 1.3.0
>   - serverinfo: 1.11.0
>   - settings: 1.3.0
>   - sharebymail: 1.11.0
>   - socialsharing_email: 2.2.0
>   - support: 1.4.0
>   - survey_client: 1.9.0
>   - systemtags: 1.11.0
>   - tasks: 0.13.6
>   - text: 3.2.0
>   - theming: 1.12.0
>   - twofactor_backupcodes: 1.10.0
>   - updatenotification: 1.11.0
>   - user_status: 1.1.1
>   - viewer: 1.5.0
>   - weather_status: 1.1.0
>   - workflowengine: 2.3.0
> ```

Hi,

Did you checked your browser's log ?
I did find informations there : in my case, it was "Uncaught TypeError: OC.Share is undefined" that led me to think that sharing app should be enabled.

**Comment by RlndVt** [comment]
With browser log do you mean the log of my reverse proxy? If so nothing interesting. My Caddyfile if that's relevant.

```
nextcloud.XXX.YY {
    root * /php-fpm-root/nextcloud/
    file_server
    php_fastcgi nc-app:9000 {
        root /var/www/html
    }
    log {
        output stderr
        output file /data/log/nextcloud.XXX.YY {
    	    roll_size 1MB
    	    roll_keep 1
    	    roll_keep_for 180h
        }
    }
    header / {
        Strict-Transport-Security           "max-age=15552000; includeSubDomains; preload;" #at least 15552000 15768000;
        X-Content-Type-Options              "nosniff"
        X-XSS-Protection                    "1; mode=block"
        X-Robots-Tag                        "none"
        X-Download-Options                  "noopen"
        X-Permitted-Cross-Domain-Policies   "none"
        Referrer-Policy                     "no-referrer"
    }
    header_up X-Forwarded-For {header.X-Forwarded-For}

    # client support (e.g. os x calendar / contacts)
    redir /.well-known/carddav /remote.php/carddav 301
    redir /.well-known/caldav /remote.php/caldav 301

    @forbidden {
        path    /.htaccess
        path    /data/*
        path    /config/*
        path    /db_structure
        path    /.xml
        path    /README
        path    /3rdparty/*
        path    /lib/*
        path    /templates/*
        path    /occ
        path    /console.php
    }

    respond @forbidden 403
}
```

**Comment by olivier6931** [comment]
> 
> 
> With browser log do you mean the log of my reverse proxy? If so nothing interesting. My Caddyfile if that's relevant.
> 
> ```
> nextcloud.XXX.YY {
>     root * /php-fpm-root/nextcloud/
>     file_server
>     php_fastcgi nc-app:9000 {
>         root /var/www/html
>     }
>     log {
>         output stderr
>         output file /data/log/nextcloud.XXX.YY {
>     	    roll_size 1MB
>     	    roll_keep 1
>     	    roll_keep_for 180h
>         }
>     }
>     header / {
>         Strict-Transport-Security           "max-age=15552000; includeSubDomains; preload;" #at least 15552000 15768000;
>         X-Content-Type-Options              "nosniff"
>         X-XSS-Protection                    "1; mode=block"
>         X-Robots-Tag                        "none"
>         X-Download-Options                  "noopen"
>         X-Permitted-Cross-Domain-Policies   "none"
>         Referrer-Policy                     "no-referrer"
>     }
>     header_up X-Forwarded-For {header.X-Forwarded-For}
> 
>     # client support (e.g. os x calendar / contacts)
>     redir /.well-known/carddav /remote.php/carddav 301
>     redir /.well-known/caldav /remote.php/caldav 301
> 
>     @forbidden {
>         path    /.htaccess
>         path    /data/*
>         path    /config/*
>         path    /db_structure
>         path    /.xml
>         path    /README
>         path    /3rdparty/*
>         path    /lib/*
>         path    /templates/*
>         path    /occ
>         path    /console.php
>     }
> 
>     respond @forbidden 403
> }
> ```

I was talking about your firefox or chrome/edge console log.
IE : press F12 on firefox to get it, and "console" tab, like this :
![image](https://user-images.githubusercontent.com/67951376/124884310-712f8d00-dfd2-11eb-99dc-5278008a1082.png)

**Comment by RlndVt** [comment]
Thanks!

```
Uncaught Error: Could not find initial state contactsinteraction of contacts
    loadState index.js:24
    <anonymous> isContactsInteractionEnabled.js:25
    Webpack 3
        o
        <anonymous>
        <anonymous>
index.js:24:10
    loadState index.js:24
    <anonymous> isContactsInteractionEnabled.js:25
    Webpack 3
        o
        <anonymous>
        <anonymous>
```

**Comment by skjnldsv** [comment]
> I'm having the same issue. For me sharing was enabled, I tried toggling without effect. Also tried enabling and disabling circles.

Hello!
I will need your browser console log to investigate this issue.
Open your console, reload your page and/or do the action leading to this issue and copy/paste the log in this thread.

<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CMD + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# IE9
1. Press F12 to open the developer tools.
2. Click the “console” tab.

# Firefox
- Press CTRL + SHIFT + K to open the Web console (COMMAND + SHIFT + K on Macs).
- or, if Firebug is installed (recommended):
    1. Press F12 to open Firebug.
    2. Click on the “console” tab.

# Opera
1. Press CTRL + SHIFT + I to open Dragonfly.
2. Click on the “console” tab.
</details>

**Comment by skjnldsv** [comment]
Moved your issue into dedicated one @RlndVt https://github.com/nextcloud/contacts/issues/2339

**Comment by Steve-CCity** [comment]
I am having this issue also.  Via web - all other tabs are fine - just contacts - completely blank.
![image](https://user-images.githubusercontent.com/67204123/126548886-abd709d8-9637-4d23-a18c-c4486d63f256.png)
![image](https://user-images.githubusercontent.com/67204123/126548940-bbe7bc2d-c723-4669-97f9-44be3e944214.png)

**Comment by skjnldsv** [comment]
For everyone reporting similar issues:

I will need your browser console log to investigate this issue.
Open your console, reload your page and/or do the action leading to this issue and copy/paste the log in this thread.

<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CMD + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# IE9
1. Press F12 to open the developer tools.
2. Click the “console” tab.

# Firefox
- Press CTRL + SHIFT + K to open the Web console (COMMAND + SHIFT + K on Macs).
- or, if Firebug is installed (recommended):
    1. Press F12 to open Firebug.
    2. Click on the “console” tab.

# Opera
1. Press CTRL + SHIFT + I to open Dragonfly.
2. Click on the “console” tab.
</details>

**Comment by Steve-CCity** [comment]
Thanks
This?
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). utils.js:35:9
Uncaught SyntaxError: missing ) after argument list

**Comment by miku86** [comment]
Nextcloud: 21.0.0
Contacts: 4.0.0

```
Uncaught TypeError: OC.Share is undefined
    87293 http://nextcloud.local/custom_apps/contacts/js/contacts-main.js?v=bb2590bd:2
    a http://nextcloud.local/custom_apps/contacts/js/contacts-main.js?v=bb2590bd:2
    46278 http://nextcloud.local/custom_apps/contacts/js/contacts-main.js?v=bb2590bd:2
    a http://nextcloud.local/custom_apps/contacts/js/contacts-main.js?v=bb2590bd:2
    <anonymous> http://nextcloud.local/custom_apps/contacts/js/contacts-main.js?v=bb2590bd:2
    <anonymous> http://nextcloud.local/custom_apps/contacts/js/contacts-main.js?v=bb2590bd:2
    <anonymous> http://nextcloud.local/custom_apps/contacts/js/contacts-main.js?v=bb2590bd:2
contacts-main.js:2:3493468
```

**Comment by aGrimRepoMan** [comment]
skjnldsv wrote:
> 
> 
> For everyone reporting similar issues:
> 
> I will need your browser console log to investigate this issue.

Here's my Firefox v90.0.2 console log.  I'm running freshly installed NC 22.0.0.1 on PMOS (Alpine Linux) with nginx 1.20.1 & mariadb 10.6.3. Sharing is enabled in Admin

```
GEThttps://<redacted>/apps/contacts/
[HTTP/2 200 OK 316ms]

< ... snip ... >

GEThttps://<redacted>/core/js/dist/unified-search.js?v=47e631d1
[HTTP/2 200 OK 268ms]

Unrecognized at-rule or error parsing at-rule “@-ms-viewport”. 5f7c-ee5e-server.css:1:23410
Unknown pseudo-class or pseudo-element “-ms-input-placeholder”.  Ruleset ignored due to bad selector. 5f7c-ee5e-server.css:1:54875
GEThttps://<redacted>/apps/files/img/app.svg?v=47e631d1
[HTTP/2 200 OK 2ms]

< ... snip ...>

GEThttps://<redacted>/core/img/logo/logo.svg?v=1
[HTTP/2 200 OK 2ms]

No OC found index.js:46:12
JQMIGRATE: Migrate is installed, version 3.3.2 jquery-migrate.min.js:2:708
Expected “none”, URL, or filter function but found “Alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:48:9
Expected “none”, URL, or filter function but found “alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:667:10
Expected “none”, URL, or filter function but found “Alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:1044:9
Expected “none”, URL, or filter function but found “Alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:1051:9
Expected “none”, URL, or filter function but found “Alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:1055:9
Expected “none”, URL, or filter function but found “Alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:1307:10
Expected “none”, URL, or filter function but found “Alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:175:9
Expected “none”, URL, or filter function but found “Alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:182:9
Expected “none”, URL, or filter function but found “Alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:186:9
Expected “none”, URL, or filter function but found “Alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:438:10
Expected “none”, URL, or filter function but found “alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:132:11
Unknown property “-moz-border-radius”.  Declaration dropped. contacts:136:22
Expected “none”, URL, or filter function but found “alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:146:11
Unknown property “-moz-border-radius”.  Declaration dropped. contacts:150:22
Unknown property “zoom”.  Declaration dropped. contacts:9:10
Expected declaration but found “*”.  Skipped to next declaration. contacts:10:5
Expected “none”, URL, or filter function but found “progid”.  Error in parsing value for “filter”.  Declaration dropped. contacts:55:13
Expected “none”, URL, or filter function but found “progid”.  Error in parsing value for “filter”.  Declaration dropped. contacts:71:13
Expected “none”, URL, or filter function but found “alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:137:13
Expected “none”, URL, or filter function but found “progid”.  Error in parsing value for “filter”.  Declaration dropped. contacts:204:13
Expected “none”, URL, or filter function but found “progid”.  Error in parsing value for “filter”.  Declaration dropped. contacts:309:13
Expected “none”, URL, or filter function but found “progid”.  Error in parsing value for “filter”.  Declaration dropped. contacts:321:13
Expected “none”, URL, or filter function but found “progid”.  Error in parsing value for “filter”.  Declaration dropped. contacts:585:13
Expected “none”, URL, or filter function but found “alpha(”.  Error in parsing value for “filter”.  Declaration dropped. contacts:15:10
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. globals.js:62:15
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. globals.js:62:15
**Uncaught TypeError: OC.Share is undefined
    <anonymous> constants.ts:128
    <anonymous> contacts-main.js:2
    <anonymous> contacts-main.js:2
constants.ts:128:16
    <anonymous> constants.ts:128
    <anonymous> contacts-main.js:2
    <anonymous> contacts-main.js:2**
Error in parsing value for “left”.  Declaration dropped. UnifiedSearch.vue:678:2
Error in parsing value for “left”.  Declaration dropped. UnifiedSearch.vue:693:4
Error in parsing value for “top”.  Declaration dropped. UnifiedSearch.vue:697:3
Error in parsing value for “top”.  Declaration dropped. UnifiedSearch.vue:720:3
Proxying an event bus of version 1.3.0 with 2.0.0 ProxyBus.ts:17:12
Loading theming data for notification bell styling App.vue:138
GEThttps://<redacted>/apps/notifications/img/notifications.svg
[HTTP/2 200 OK 2ms]

$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. globals.js:62:15
Registering notifications container as a menu App.vue:148
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 2 globals.js:62:15
Notifications permissions granted App.vue:364
Polling interval updated to 30000 App.vue:325
Did not start background fetcher as session_keepalive is off App.vue:207
XHRGEThttps://<redacted>/ocs/v2.php/apps/notifications/api/v2/notifications
[HTTP/2 200 OK 203ms]

Error in parsing value for “left”.  Declaration dropped. UnifiedSearch.vue:673:2
Error in parsing value for “left”.  Declaration dropped. UnifiedSearch.vue:688:3
Error in parsing value for “top”.  Declaration dropped. UnifiedSearch.vue:697:3
Error in parsing value for “top”.  Declaration dropped. UnifiedSearch.vue:711:2
session heartbeat disabled session-heartbeat.js:149:10
XHRGEThttps://<redacted>/ocs/v2.php/search/providers?from=/apps/contacts/
[HTTP/2 200 OK 341ms]

Got notification data App.vue:288
Polling interval updated to 30000 App.vue:325
Did not start background fetcher as session_keepalive is off App.vue:207
GEThttps://<redacted>/apps-appstore/contacts/img/favicon-touch.png
[HTTP/2 200 OK 2ms]

GEThttps://<redacted>/apps-appstore/contacts/img/favicon.ico
[HTTP/2 200 OK 2ms]

```

Sharing is configured as per screenshot:
![Screenshot 2021-07-29 at 20-19-25 Settings - Nextcloud](https://user-images.githubusercontent.com/34463752/127560431-85f2e5f6-9c12-4c69-9263-cb45bcb6b97b.png)

**Comment by Steve-CCity** [comment]
@skjnldsv 
> For everyone reporting similar issues:
> 
> I will need your browser console log to investigate this issue.
> Open your console, reload your page and/or do the action leading to this issue and copy/paste the log in this thread.
> How to access your browser console (Click to expand)

Any ideas on what I can do to make my contacts visible?  How long it might take to figure out what is going wrong?
Thanks

**Comment by beerisgood** [comment]
Edge  92.0.902.62
Nextcloud 21.0.3

> index.js:46 No OC found
Fr @ index.js:46
value @ gettext.js:45
(anonym) @ l10n.js:3
(anonym) @ main.js?v=d8aea090-0:1
n @ bootstrap:19
(anonym) @ main.js?v=d8aea090-0:1
n @ bootstrap:19
(anonym) @ main.js:1
(anonym) @ main.js?v=d8aea090-0:1
n @ bootstrap:19
(anonym) @ bootstrap:83
(anonym) @ main.js?v=d8aea090-0:1
jquery-migrate.min.js:2 JQMIGRATE: Migrate is installed, version 3.3.2
globals.js:60 jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own.
pe @ globals.js:60
get @ globals.js:91
(anonym) @ jquery.js:10336
(anonym) @ jquery.js:28
0 @ jquery.js:14
n @ bootstrap:19
706 @ files_client.js?v=d8aea090-0:1
n @ bootstrap:19
(anonym) @ bootstrap:83
(anonym) @ files_client.js?v=d8aea090-0:1
globals.js:60 $ is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own.
pe @ globals.js:60
get @ globals.js:91
(anonym) @ jquery.js:10339
(anonym) @ jquery.js:28
0 @ jquery.js:14
n @ bootstrap:19
706 @ files_client.js?v=d8aea090-0:1
n @ bootstrap:19
(anonym) @ bootstrap:83
(anonym) @ files_client.js?v=d8aea090-0:1
globals.js:60 $ is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own.
pe @ globals.js:60
get @ globals.js:91
mounted @ App.vue:121
zt @ vue.runtime.esm.js:1854
Ke @ vue.runtime.esm.js:4219
insert @ vue.runtime.esm.js:3139
L @ vue.runtime.esm.js:6346
(anonym) @ vue.runtime.esm.js:6565
t._update @ vue.runtime.esm.js:3945
r @ vue.runtime.esm.js:4066
pn.get @ vue.runtime.esm.js:4479
pn @ vue.runtime.esm.js:4468
t @ vue.runtime.esm.js:4073
Mn.$mount @ vue.runtime.esm.js:8415
t._init @ vue.runtime.esm.js:5018
Mn @ vue.runtime.esm.js:5085
(anonym) @ main.js:32
n @ bootstrap:19
(anonym) @ bootstrap:83
(anonym) @ notifications-main.js?v=d8aea090-0:1
globals.js:60 $ is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own.
pe @ globals.js:60
get @ globals.js:91
mounted @ App.vue:125
zt @ vue.runtime.esm.js:1854
Ke @ vue.runtime.esm.js:4219
insert @ vue.runtime.esm.js:3139
L @ vue.runtime.esm.js:6346
(anonym) @ vue.runtime.esm.js:6565
t._update @ vue.runtime.esm.js:3945
r @ vue.runtime.esm.js:4066
pn.get @ vue.runtime.esm.js:4479
pn @ vue.runtime.esm.js:4468
t @ vue.runtime.esm.js:4073
Mn.$mount @ vue.runtime.esm.js:8415
t._init @ vue.runtime.esm.js:5018
Mn @ vue.runtime.esm.js:5085
(anonym) @ main.js:32
n @ bootstrap:19
(anonym) @ bootstrap:83
(anonym) @ notifications-main.js?v=d8aea090-0:1
globals.js:60 $ is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own.
pe @ globals.js:60
get @ globals.js:91
mounted @ App.vue:125
zt @ vue.runtime.esm.js:1854
Ke @ vue.runtime.esm.js:4219
insert @ vue.runtime.esm.js:3139
L @ vue.runtime.esm.js:6346
(anonym) @ vue.runtime.esm.js:6565
t._update @ vue.runtime.esm.js:3945
r @ vue.runtime.esm.js:4066
pn.get @ vue.runtime.esm.js:4479
pn @ vue.runtime.esm.js:4468
t @ vue.runtime.esm.js:4073
Mn.$mount @ vue.runtime.esm.js:8415
t._init @ vue.runtime.esm.js:5018
Mn @ vue.runtime.esm.js:5085
(anonym) @ main.js:32
n @ bootstrap:19
(anonym) @ bootstrap:83
(anonym) @ notifications-main.js?v=d8aea090-0:1
constants.ts:128 Uncaught TypeError: Cannot read property 'SHARE_TYPE_USER' of undefined
    at constants.ts:128
    at contacts-main.js?v=d8aea090-0:2
    at contacts-main.js?v=d8aea090-0:2
(anonym) @ constants.ts:128
(anonym) @ contacts-main.js?v=d8aea090-0:2
(anonym) @ contacts-main.js?v=d8aea090-0:2
globals.js:60 jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own.
pe @ globals.js:60
get @ globals.js:91
(anonym) @ script.js?v=d8aea090-0:492
globals.js:60 jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own.
pe @ globals.js:60
get @ globals.js:91
(anonym) @ files.js?v=d8aea090-0:122
session-heartbeat.js:97 session heartbeat polling started
globals.js:60 $ is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own.
pe @ globals.js:60
get @ globals.js:91
displayEncryptionWarning @ encryption.js?v=d8aea090-0:18
(anonym) @ encryption.js?v=d8aea090-0:32
(anonym) @ delay.js:7
setTimeout (asynchron)
(anonym) @ delay.js:6
(anonym) @ restArguments.js:18
Xe @ _executeBound.js:8
i @ partial.js:18
(anonym) @ encryption.js?v=d8aea090-0:31

**Comment by eric-pierce** [comment]
I have this issue as well - enabling "File Sharing" does appear to resolve, and when I do so it doesn't throw the "Uncaught TypeError: OC.Share is undefined" error.

Details of my config without "File Sharing" enabled below:

Contacts version 4.0.1
Nextcloud (Docker) version 22.0.0

`Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). moz-extension:35:9
No OC found index.js:46:12
JQMIGRATE: Migrate is installed, version 3.3.2 jquery-migrate.min.js:2:708
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. globals.js:62:15
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. globals.js:62:15
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 2 globals.js:62:15
Uncaught TypeError: OC.Share is undefined
    <anonymous> constants.ts:128
    <anonymous> contacts-main.js:2
    <anonymous> contacts-main.js:2
constants.ts:128:16
Proxying an event bus of version 1.3.0 with 2.0.0 ProxyBus.ts:17:12
Loading theming data for notification bell styling App.vue:138
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. globals.js:62:15
Registering notifications container as a menu App.vue:148
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 2 globals.js:62:15
Notifications permissions granted App.vue:364
Polling interval updated to 30000 App.vue:325
Started background fetcher as session_keepalive is enabled App.vue:204
session heartbeat polling started session-heartbeat.js:101:9
Got notification data App.vue:288
Polling interval updated to 30000`

**Comment by Steve-CCity** [comment]
Enabling File Sharing? How is that done?
I do not host my own Nextcloud, I am using a service - Tab Digital and cannot find where to do it

**Comment by eric-pierce** [comment]
> 
> 
> Enabling File Sharing? How is that done?
> I do not host my own Nextcloud, I am using a service - Tab Digital and cannot find where to do it

You can enable it under the "Apps" configuration page available to administrators:
![image](https://user-images.githubusercontent.com/551464/128245546-b77b036b-f4ab-4790-85f6-fc179600781c.png)

**Comment by Steve-CCity** [comment]
I cannot find this. Here is what I see - 
![image](https://user-images.githubusercontent.com/67204123/128247790-010c0fa4-4cc8-430d-8642-4ec8afc06e32.png)

**Comment by eric-pierce** [comment]
> 
> 
> I cannot find this. Here is what I see -
> ![image](https://user-images.githubusercontent.com/67204123/128247790-010c0fa4-4cc8-430d-8642-4ec8afc06e32.png)

The "Apps" screen is not under settings - you can find it by clicking on your profile circle in the top right of the screen and selecting "Apps". It looks like you may not be on an administrator account as I don't see administrator settings in your settings screenshot - you'll need to reach out to whoever the administrator of your Nextcloud instance to have them enable the file sharing app.

**Comment by ChristophWurst** [comment]
Fix is at https://github.com/nextcloud/contacts/pull/2400

**Comment by ChristophWurst** [comment]
v4.0.2 released.

**Comment by szaimen** [comment]
🎉

**Comment by g-d-d-h** [comment]
I am having this issue. Nextcloud and Apps are saying everything is up to date.

**Comment by RoelSG** [comment]
I am also having this issue, apps are also all up to date.

**Comment by alpacathunder1** [comment]
Same issue here, thought it was an Nginx thing at first, but it appears to be unrelated...

**Comment by brrrrrrrt** [comment]
problem still occurs with v4.0.4 (nextcloud 22.2.3) 

Javascript console tells:
Refused to execute script from 'https://my.domain.com/apps/contacts/js/contacts-main.js?v=c87f9709-11' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.

When i open "https://my.domain.com/apps/contacts/js/contacts-main.js?v=c87f9709-11" it just opens the same empty page, no javascript at all.

**Comment by itsthejb** [comment]
@brrrrrrrt I still have 4.0.4, but reinstalling fixed it for me

**Comment by wadouk** [comment]
@brrrrrrrt @itsthejb how do you do that, reinstalling ? removing and install does it drop database ?
same here, doesn't offer to upgrade

**Comment by itsthejb** [comment]
Uninstall, install again. It’s safe

**Comment by laigor** [comment]
> Uninstall, install again. It’s safe

In my case, this recommendation solved the problem.

**Comment by simon-mueller** [comment]
Completely removing (`Disable`, then `Remove`) and then `Download & Enable` did the job finally for me. What a relief!

**Comment by killdashnine420** [comment]
Hello, wanted to make a quick note of my personal experience in case anyone else runs into it because it was a pain in the ass to find the resource.

NextCloud version: 30.0.4
Contacts version: 6.1.1

I had to go to my Nginx config file and manually enter ".mjs" into the application/javascript section due to Nginx not automatically serving .mjs files using JavaScript MIME type. I used the console in Firefox to find the error and followed the trail.

This NextCloud instance is on a TrueNAS server.

Restarted the NextCloud server and BAM! Contacts table appeared again.

Hope this helps anyone in the future! :)
